:bug:, :recycle: [cmd, demo] fix error on cmd help

matthiasgeihs opened the pull request:
- fixed error caused by trying to read config files when executing cli help
- moved demo command definition to package demo to allow for accessing cli variables

Fixes #15 

The problem was that when running the help command, it already tried to load the config files, where "config.yaml" is not present. My first idea was to load the config files only when running the demo command. However, for that I needed to make the config file parameters available to the package demo. Unfortunately, the demo command definition for some reason was in package cmd. To make it work, I refactored the demo command definition and its tests into package demo (where I think it belongs anyways).

ggwpez commented:
If you resolve an issue with a MR, then use the *go-perun* naming scheme for the merge Branch (here: `15-help-error`) so we see to which issue a branch correlates.  
The `fix/*` notation is for stuff that does not have/need an issue.

matthiasgeihs commented:
> If you resolve an issue with a MR, then use the _go-perun_ naming scheme for the merge Branch (here: `15-help-error`) so we see to which issue a branch correlates.
> The `fix/*` notation is for stuff that does not have/need an issue.

ok, so do you want me to rename the branch and create a new PR, or is it ok if i stick to the naming scheme next time?

ggwpez commented:
> ok, so do you want me to rename the branch and create a new PR, or is it ok if i stick to the naming scheme next time?

It is fine for now, just so you know for the future :smile:

ggwpez commented on cmd/demo/demo.go:
```diff
@@ -25,21 +23,28 @@ var demoCmd = &cobra.Command{
 }
 
-var testAPI bool
+var testAPIFlag bool
```
Maybe create another struct `demoConfig` which holds `testAPIEnabled` (rename from `testAPIFlag`), `cfgFile` and `cfgNetFile`. Just like the `logCfg` in `cmd/config.go`.

matthiasgeihs replied:
ok, i created a struct for the command line flags.

ggwpez commented on cmd/demo/demo.go:
```diff
@@ -25,21 +23,28 @@ var demoCmd = &cobra.Command{
 	viper.BindPFlag("secretkey", demoCmd.PersistentFlags().Lookup("sk"))
 }
 
+// GetDemoCmd exposes demoCmd so that it can be used as a sub-command by another cobra command instance
```
```suggestion
// GetDemoCmd exposes demoCmd so that it can be used as a sub-command by another cobra command instance.
```

matthiasgeihs replied:
done

ggwpez commented on cmd/root.go:
```diff
@@ -12,42 +12,27 @@ import (
 	PersistentPreRun: runRoot,
 }
 
 var cfgFile, cfgNetFile string
```
```suggestion
```

ggwpez replied:
Why is this resolved?

matthiasgeihs replied:
my bad, done now.

ggwpez commented on cmd/demo/demo.go:
```diff
@@ -25,21 +23,28 @@ var demoCmd = &cobra.Command{
+var testAPIFlag bool
+var cfgFile, cfgNetFile string
 
 func init() {
```
Thinking about it, we can probably put this whole function into `cmd/demo/Config.go`, (same for `root.go`) which would make it a bit easier to under stand since all the configuration is then in these two files.

ggwpez replied:
You did not like this :rofl: ?

matthiasgeihs replied:
by the whole function you mean `func init()`? i thought init fits well in file demo.go, as the package is also called "demo" and hence this is where you might expect the init function to be located. furthermore, since init initializes the command line flags, i thought it made sense to keep them within that same file. however, i don't care too much. both ways (keep in demo.go or move to config.go) are fine for me. you decide.

ggwpez replied:
Then its ok.

matthiasgeihs committed:
:bug:, :recycle: [cmd, demo] fix error on cmd help

- fixed error caused by trying to read config files when executing cli help
- moved demo command definition to package demo to allow for accessing cli variables
- added command line flag struct

ggwpez commented:
<s>It still fails for me when i do `go build . && perun-eth-demo help` on the `fix-command-help` branch…  </s>
PS: works

ggwpez reviewed (changes requested):
Ok looks great.  
Can you please squash 84b200516c310a015a728e03f496f5cbcd891bf0 and 7987932f734c4759ff40ffb9f63daf2c4867bf5d into something like ":recycle: Restructure config variables"?

matthiasgeihs commented:
> Ok looks great.
> Can you please squash [84b2005](https://github.com/perun-network/perun-eth-demo/commit/84b200516c310a015a728e03f496f5cbcd891bf0) and [7987932](https://github.com/perun-network/perun-eth-demo/commit/7987932f734c4759ff40ffb9f63daf2c4867bf5d) into something like "♻️ Restructure config variables"?

squashed everything into one commit since [7987932](https://github.com/perun-network/perun-eth-demo/commit/7987932f734c4759ff40ffb9f63daf2c4867bf5d) was a fixup of the initial commit. ok?
